src/as-validator.c

```diff
@@ -1228,20 +1228,27 @@ as_validator_check_tags (AsValidator *validator, xmlNode *node)
 static void
 as_validator_check_screenshots (AsValidator *validator, xmlNode *node, AsComponent *cpt)
 {
+	gboolean default_screenshot_found = FALSE;
 	for (xmlNode *iter = node->children; iter != NULL; iter = iter->next) {
 		xmlNode *iter2;
 		gboolean image_found = FALSE;
 		gboolean video_found = FALSE;
 		gboolean caption_found = FALSE;
 		gboolean default_screenshot = FALSE;
 		g_autofree gchar *scr_kind_str = NULL;
+		gboolean default_source_found = FALSE;
+		g_autoptr(GHashTable) known_source_lang = NULL;
+
+		known_source_lang = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
 
 		if (iter->type != XML_ELEMENT_NODE)
 			continue;
 
 		scr_kind_str = as_xml_get_prop_value (iter, "type");
-		if (g_strcmp0 (scr_kind_str, "default") == 0)
+		if (g_strcmp0 (scr_kind_str, "default") == 0) {
```

Comment:
Please don't do this. Use `as_screenshot_kind_from_string` and check whether `as_screenshot_kind_from_string (scr_kind_str) == AS_SCREENSHOT_KIND_DEFAULT` (more consistent, easier for refactoring, if you get an `UNKNOWN` you can already raise a validation warning.